packages/api/src/praise/utils/core.ts

```diff
@@ -86,33 +86,47 @@ export const prepareDiscordClient = async (): Promise<Client> => {
   });
   await discordClient.login(process.env.DISCORD_TOKEN);
   await discordClient.guilds.fetch();
-  const discordGuild = await discordClient.guilds.fetch(
-    process.env.DISCORD_GUILD_ID as string
-  );
-  await discordGuild.members.fetch();
-
   return discordClient;
 };
 
 /**
  * Convert text from discord into a "realized" form
  *  replacing raw references to channels and users with their human-readable text
  * @param discordClient
- * @param discordChannelId
  * @param text
  * @returns
  */
 export const realizeDiscordContent = async (
   discordClient: Client,
-  discordChannelId: string,
   text: string
 ): Promise<string> => {
-  const channel = await discordClient.channels.fetch(discordChannelId);
-
-  if (!channel) throw Error('Failed to fetch channel from discord api');
-  if (!channel.isText()) throw Error('Channel must be a TextChannel');
-
-  const textRealized = Util.cleanContent(text, channel);
-
+  const discordGuild = await discordClient.guilds.fetch(
+    process.env.DISCORD_GUILD_ID as string
+  );
+  if (!discordGuild) throw Error('Failed to fetch guild from discord api');
+  await discordGuild.members.fetch();
+  await discordGuild.channels.fetch();
```

Comment:
Do we need to do all these fetches from the discord api on every call to `realizeDiscordClient`? 

Since the `DISCORD_GUILD_ID` is static, wouldn't the fetched members and channels be the same (at least for calls m? 

We'll need to be able to run `realizeDiscordClient()` many times in a batch (such as when we're importing a bunch of praise, or in a migration). 

Currently, when I tried to create a new migration that calls this function for every praise.reason, it runs for about 2 minutes and then fails due a time out from discord api. I'd imaging I'm running to a rate limit or something.

Reply:
Yeah I think that makes sense -- try to use the value in the cache, and it fails, query the api.

Reply:
there's a small issue .replace doesn't support async functions.
Should we implement our own implementation that supports async-await, or should we use the `.then().catch()` syntax?

__
[![sema-logo](https://app.semasoftware.com/img/sema-tray-logo.gif)](https://semasoftware.com/gh) &nbsp;**Summary:** :hammer_and_wrench: This code needs a fix

Reply:
> there's a small issue .replace doesn't support async functions. Should we implement our own implementation that supports async-await, or should we use the `.then().catch()` syntax?

Could we collect the list of user names, then confirm we have them cached in the discordClient or fetch them, then run the `.replace` since we would know we had all the data already?

Reply:
this could lead to outdated data being used when the bot's `/praise` or `/forward` commands are used.

__
[![sema-logo](https://app.semasoftware.com/img/sema-tray-logo.gif)](https://semasoftware.com/gh) &nbsp;**Tags:** Not reusable